### ISSUE TITLE
[Not-Issue] How did you embed these so cool interactive charts on your blog?

### ISSUE THREAD

**Issue by kuatroka** [lukesalamone/lukesalamone.github.io#2]
Hi Luke,

I must say, your blog is impressive. The aesthetic appeal of your visualisations is so captivating that I find myself lingering on your posts, playing with the interactive charts, sliders, and filters, rather than just reading the blog itself.

I do have a question, however. I understand that GitHub Pages can only host static websites, so I'm curious about how you manage to embed these interactive features into your posts, and how you access data with those charts. For example, the hover effect on this post (https://lukesalamone.github.io/posts/self-attention/), the interactive/filter label in this chart (https://lukesalamone.github.io/posts/bert-vs-gpt2/), the slider and autoplay on this post (https://lukesalamone.github.io/posts/how-does-convolution-work/), and that slider on this one (https://lukesalamone.github.io/posts/what-is-temperature/).

Could you perhaps recommend a tutorial on how to do this? I'm interested in building an app with a lot of charts, graphs, and analytics, but I'd like to start with something simple like a static webpage and just experiment with the metrics, layout, or chart types. I've considered using Quatro (https://quarto.org/), where I can build static pages with charts directly from Jupyter Lab, but your approach seems much more elegant.

Thank you for your time and expertise.

**Comment by lukesalamone** [comment]
Hi,

Thank you for the kind words. I’m glad someone likes the visuals as much as
I did when I created them!

However I have to warn you that there is no easy route to creating the
visuals. I usually use graph.js but that library is by no means perfect.
Typically the flow will involve storing data in a static json file which
can be pulled in as the page loads, parsing it if necessary, and loading it
into Graph.js.

My philosophy is that the best way to learn to do something is to do it. So
if you want to learn to use a library like graph.js I would recommend just
getting your hands dirty with a simple example from their site, then adding
features as you see fit.

~ Luke
